src/main/scala/fs/fusejna.scala

```diff
@@ -0,0 +1,114 @@
+package suffuse
+package fs
+
+import net.fusejna._
+import jio._
+import scala.util.Properties.isMac
+
+/** Cleaning up the javacentric fuse-jna types.
+ */
+
+object Node {
+  import net.fusejna.types.TypeMode.NodeType
+
+  final val BlockDev = NodeType.BLOCK_DEVICE
+  final val Dir      = NodeType.DIRECTORY
+  final val Fifo     = NodeType.FIFO
+  final val File     = NodeType.FILE
+  final val Link     = NodeType.SYMBOLIC_LINK
+  final val Socket   = NodeType.SOCKET
+}
+
+/** Widening access so we don't have to use inheritance everywhere.
+ */
+
+trait FuseFs extends FuseFilesystem {
+  def unmountTry(): Unit = (
+    if (!isMounted)
+      return
+    else if (isMac)
+      exec("umount", "-f", getMountPoint.getPath) orElse exec("diskutil", "unmount", getMountPoint.getPath)
+    else
+      exec("fusermount", "-u", getMountPoint.getPath)
+  )
+  private def doMount(mountPoint: File, blocking: Boolean): this.type = {
+    addUnmountHook(this)
+    super.mount(mountPoint, blocking)
+    this
+  }
+
+  def mountBg(mountPoint: Path): this.type = doMount(mountPoint.toFile, false)
```

Comment:
Blocking in this context is a synonym for foreground. It means the call to mount won't return until the filesystem is unmounted, whereas in the background version it returns immediately.

If we didn't have the background version we'd have to reimplement it. For instance a command-line suffuse-mount would start everything backgrounded, just like the real mount does.